models/user.go

```diff
@@ -520,7 +520,11 @@ func (u *User) UploadAvatar(data []byte) error {
 	}
 
 	u.UseCustomAvatar = true
-	u.Avatar = fmt.Sprintf("%x", md5.Sum(data))
+	// Different users can upload same image as avatar
+	// If we prefix it with u.ID, it will be separated
+	// Otherwise, if any of the users delete his avatar
+	// Other users will lose their avatars too.
+	u.Avatar = fmt.Sprintf("%x", md5.Sum([]byte(fmt.Sprintf("%d-%x", u.ID, md5.Sum(data)))))
```

Comment:
Different user can be add same image but its low probability. Instead, same user can use same image in multi time.
So what will happen if user add same image to organizations and remove one organization?
Or add same image for own avatar and own repos' avatars?

Reply:
> Different user can be add same image but its low probability. Instead, same user can use same image in multi time.

We are using hash value of the user uploaded image prefixed with userID. So, if same user uploads same photo multiple time as his avatar, it will generate same hash value throughout the time. As we are prefixing the hash value with user id, the avatar name will be always same for same photo & same user.

> So what will happen if user add same image to organizations and remove one organization?

As I mentioned we are prefixing user id with the photo hash, it will not cause any problem. Organization is also like a user having different id. Every organization has different id, so it will not replace the previous image or delete other organization's avatar while deleting another organization.

> Or add same image for own avatar and own repos' avatars then delete one repo?

Repos' avatars are stored in separate directory and also they are prefixing with repo id with image hash which will not conflict with one another.
Even if both avatars are stored in same directory, they will not collide as for user avatar we are taking another hash value of `<hash>[<userid>-<image_hash>]` and for repo the final hash isn't using.

Hope I made myself clear.